ng-dev/pr/checkout/checkout.ts

```diff
@@ -0,0 +1,101 @@
+import {dirname, join} from 'path';
+import {AuthenticatedGitClient} from '../../utils/git/authenticated-git-client.js';
+import {Prompt} from '../../utils/prompt.js';
+import {Log, bold, green} from '../../utils/logging.js';
+import {checkOutPullRequestLocally} from '../common/checkout-pr.js';
+import {fileURLToPath} from 'url';
+
+/** List of accounts that are supported for takeover. */
+const takeoverAccounts = ['angular-robot'];
+
+export interface CheckoutPullRequestParams {
+  pr: number;
+  takeover?: boolean;
+}
+
+export async function checkoutPullRequest(params: CheckoutPullRequestParams): Promise<void> {
+  const {pr, takeover} = params;
+  /** An authenticated git client. */
+  const git = await AuthenticatedGitClient.get();
+  /** The branch name used for the takeover change. */
+  const branchName = `pr-takeover-${pr}`;
+
+  // Make sure the local repository is clean.
+  if (git.hasUncommittedChanges()) {
+    Log.error(
+      ` ✘ Local working repository not clean. Please make sure there are no uncommitted changes`,
+    );
+    return;
+  }
+
+  const {resetGitState, pullRequest, pushToUpstreamCommand} = await checkOutPullRequestLocally(pr, {
+    allowIfMaintainerCannotModify: true,
+  });
+
+  // if maintainer can modify is false or if takeover is provided do takeover
+
+  if (pullRequest.maintainerCanModify === false || takeover) {
+    if (takeover !== true) {
+      Log.info('The author of this pull request does not allow maintainers to modify the pull');
+      Log.info('request. Since you will not be able to push changes to the original pull request');
+      Log.info('you will instead need to perform a "takeover." In a takeover the original pull');
+      Log.info('request will be checked out, the commits are modified to close the original on');
+      Log.info('merge of the newly created branch.\n');
+
+      if (!(await Prompt.confirm(`Would you like to create a takeover pull request?`, true))) {
+        Log.info('Aborting takeover..');
+        await resetGitState();
+        return;
+      }
+    }
+
+    if (git.runGraceful(['rev-parse', '-q', '--verify', branchName]).status === 0) {
```

Comment:
Nit: Consider extracting this block (the whole logic inside takeover) into a file/function

Reply:
I think that eventually it might make sense, but going to leave it for now